circle.yml

```diff
@@ -1,41 +1,99 @@
-general:
-  artifacts:
-  - "target/test.log"
-  - "gc.log"
+defaults:
+  - &defaults
+    working_directory: ~/kafka-serialisation
+    docker:
+      - image: olib963/sbt-ci-image:sbt-1-1-5
+  - &dependency-cache v2-dependencies-{{ checksum "build.sbt" }}-{{ checksum "./build/coursier.sh" }}
+  - &restore-cache
+    restore_cache:
+      keys:
+        - *dependency-cache
+        # fallback to using the latest cache if no exact match is found
+        - v2-dependencies-
+  - &install-coursier
+    run:
+      name: Install Coursier
+      command: ./build/coursier.sh
 
-machine:
-  environment:
-    JAVA_OPTS: "-XX:+CMSClassUnloadingEnabled -XX:MaxMetaspaceSize=512M -XX:MetaspaceSize=512M -Xms1G -Xmx1G -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -Xloggc:gc.log"
-    TEST_TIMEFACTOR: 5.0
-  java:
-    version: oraclejdk8
+version: 2
+jobs:
+  build:
+    <<: *defaults
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - *restore-cache
+      - *install-coursier
+      - run:
+          name: Change test concurrency # TODO why is this not just in the repo?
```

Comment:
That's a good question, @filosganga?

Reply:
The circleci machine executor has very limited resources. While locally we can parallelize more on circleci v1 we could not. 

On the v2 we may try to remove this limit.

Reply:
Shall I leave it here for now then or just remove it and see what happens?